tests/main/interfaces-system-observe/task.yaml

```diff
@@ -10,39 +13,66 @@ details: |
 
 prepare: |
     echo "Given a snap declaring a plug on the system-observe interface is installed"
-    snapbuild $TESTSLIB/snaps/system-observe-consumer .
-    snap install --dangerous system-observe-consumer_1.0_all.snap
+    snap install --edge test-snapd-system-observe-consumer
+
+    echo "And the D-bus X11 dependencies are installed"
+    apt install -y dbus-x11
+
+    if [[ "$SPREAD_SYSTEM" != ubuntu-14.04-* ]]; then
```

Comment:
You have 4 places that do the same check which to me suggests using a variable (you could name the variable something "14.04-y" if you like which of course is a small maintenance cost if this is to apply to anything else). To be clear, I'm not blocking on this point.